front/src/reducers/infra/index.ts

```diff
@@ -2,20 +2,26 @@ import type { CaseReducer, Draft, PayloadAction } from '@reduxjs/toolkit';
 
 export interface InfraState {
   infraID: number | undefined;
+  infraLockStatus: boolean;
 }
 
 export const infraState: InfraState = {
   infraID: undefined,
+  infraLockStatus: false,
 };
 
 export interface InfraStateReducers<S extends InfraState> {
   ['updateInfraID']: CaseReducer<S, PayloadAction<S['infraID']>>;
+  ['updateInfraLockStatus']: CaseReducer<S, PayloadAction<S['infraLockStatus']>>;
 }
 
 export function buildInfraStateReducers<S extends InfraState>(): InfraStateReducers<S> {
   return {
     updateInfraID(state: Draft<S>, action: PayloadAction<S['infraID']>) {
       state.infraID = action.payload;
     },
+    updateInfraLockStatus(state: Draft<S>, action: PayloadAction<S['infraLockStatus']>) {
+      state.infraLockStatus = action.payload;
+    },
```

Comment:
We should update the lockStatus at the same time than the id to avoid any inconsistencies. You can add this in updateInfraID and remove updateInfraLockStatus
```suggestion
    updateInfraID(state: Draft<S>, action: PayloadAction<Infra>) {
      state.infraID = action.payload.id;
      state.infraLockStatus = action.payload.locked;
    },
```